k256/src/arithmetic/affine.rs

```diff
@@ -186,16 +197,22 @@ impl DecompressPoint<Secp256k1> for AffinePoint {
                     beta.is_odd().ct_eq(&y_is_odd),
                 );
 
-                Self {
-                    x,
-                    y: y.normalize(),
-                    infinity: 0,
-                }
+                Self::new(x, y.normalize())
             })
         })
     }
 }
 
+/// Decompaction using Taproot conventions as described in [BIP 340].
+///
+/// [BIP 340]: https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki
+impl DecompactPoint<Secp256k1> for AffinePoint {
+    fn decompact(x_bytes: &FieldBytes) -> CtOption<Self> {
+        // TODO(tarcieri): implement full `lift_x` algorithm as described in BIP 340
```

Comment:
Still an important TODO here

Reply:
As mentioned in the other PR, I'm missing what the remaining TODO is, so removing the marker. FieldElement::sqrt calculates y via the (p+1)/4'th power, checking it's an actual square root.